File: qc-core/src/main/java/com/eager/questioncloud/core/domain/subscribe/model/Subscribe.java
package com.eager.questioncloud.core.domain.subscribe.model;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
public class Subscribe {
    private Long id;
    private Long subscriberId;
    private Long creatorId;
    private LocalDateTime createdAt;

    @Builder
    public Subscribe(Long id, Long subscriberId, Long creatorId, LocalDateTime createdAt) {
        this.id = id;
        this.subscriberId = subscriberId;
        this.creatorId = creatorId;
        this.createdAt = createdAt;
    }

    public static Subscribe create(Long subscriberId, Long creatorId) {
        return Subscribe.builder()
            .subscriberId(subscriberId)
            .creatorId(creatorId)
            .createdAt(LocalDateTime.now())
            .build();
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/subscribe/model/Subscribe.kt
package com.eager.questioncloud.core.domain.subscribe.model

import java.time.LocalDateTime

class Subscribe(
    val id: Long = 0,
    val subscriberId: Long,
    val creatorId: Long,
    val createdAt: LocalDateTime = LocalDateTime.now(),
) {
    companion object {
        fun create(subscriberId: Long, creatorId: Long): Subscribe {
            return Subscribe(subscriberId = subscriberId, creatorId = creatorId)
        }
    }
}
